### PR TITLE
arch/armv7-m/up_assert: Add abort handling in fault manager for armv7-m

### PR DESCRIPTION
**Pull request by drashti304**
This patch adds support of abort handling in fault manger for armv7-m architecture.

All kinds of hard faults is handled by fault manager

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [a1821b4ea62f1e17c10796a37cccf8f00a922a54] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a1821b4ea62f1e17c10796a37cccf8f00a922a54] - Code Rule Check OK.

**Comment by jeongarmy** [comment]
please update commit message for binary_manager

File: os/arch/arm/src/armv7-m/up_assert.c [hunk 1]
```diff
@@ -491,7 +507,7 @@ void dump_all_stack(void)
  * Name: up_assert
  ****************************************************************************/
 
-void up_assert(const uint8_t *filename, int lineno)
+int up_assert(const uint8_t *filename, int lineno)
```
Comment (sunghan-chang) [inline os/arch/arm/src/armv7-m/up_assert.c]: I don't know why is needed? You only return OK

Comment (sangam-swami) [reply os/arch/arm/src/armv7-m/up_assert.c]: yes done with PR #3017

**Comment by sangam-swami** [comment]
Will be followed in PR # 3017

**Pull request closed without merging**